extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -72,24 +71,47 @@ public class DefaultTenantConfigResolver {
     boolean enableHttpForwardedPrefix;
 
     DefaultTenantConfigResolver(BlockingSecurityExecutor blockingExecutor, BeanManager beanManager,
+            Instance<TenantResolver> tenantResolver,
             @ConfigProperty(name = "quarkus.security.events.enabled") boolean securityEventsEnabled,
+            @ConfigProperty(name = "quarkus.oidc.select-tenants-with-issuer") boolean selectTenantsWithIssuer,
             @ConfigProperty(name = "quarkus.http.root-path") String rootPath, TenantConfigBean tenantConfigBean) {
         this.blockingRequestContext = new BlockingTaskRunner<OidcTenantConfig>(blockingExecutor);
         this.securityEventObserved = SecurityEventHelper.isEventObserved(new SecurityEvent(null, (SecurityIdentity) null),
                 beanManager, securityEventsEnabled);
         this.tenantConfigBean = tenantConfigBean;
-        this.pathMatchingTenantResolver = PathMatchingTenantResolver.of(tenantConfigBean.getStaticTenantsConfig(), rootPath,
+
```

Comment:
I am adding yet another resolver. It has impacts - it is another IF condition. Therefore it is not arbitrary refactoring, it is keeping code clean, simple and effective.

Reply:
My point is, whether you call it refactoring or not, it is kind of change that should be done as part of the PR because you can just add code without considering it's impact.

Of course, there must be agreement on every change and I can understand if you disagree.

Reply:
I thought about this over night, and truth be told, I made these sort of changes in almost every PR I do. And in these where I don't it's usually because I don't feel responsibility of the code and just try to patch thing :-) I think it's important to try keep code quality in general.

If you are certain this change is wrong, I'll drop it.

Reply:
@michalvavrik Let me comment below